cmake/projects/libsodium/schemes/url_sha1_libsodium_msbuild.cmake.in

```diff
@@ -0,0 +1,129 @@
+string(COMPARE EQUAL "@HUNTER_MSVC_VERSION@" "11" _is_platform_toolset_v110)
+string(COMPARE EQUAL "@HUNTER_MSVC_VERSION@" "12" _is_platform_toolset_v120)
+string(COMPARE EQUAL "@HUNTER_MSVC_VERSION@" "14" _is_platform_toolset_v140)
+
+if(_is_platform_toolset_v100)
+    set(msvc_platform_toolset "v100")
+elseif(_is_platform_toolset_v110)
+    set(msvc_platform_toolset "v110")
+elseif(_is_platform_toolset_v120)
+    set(msvc_platform_toolset "v120")
+elseif(_is_platform_toolset_v140)
+    set(msvc_platform_toolset "v140")
+else()
+    hunter_user_error("Visual studio version not supported. Supported versions are: Visual studio 2010, 2011, 2012 and 2014.")
+    hunter_user_error("If there is a new version please open an issue at https://github.com/ruslo/hunter and mention @Cyberunner23")
+endif()
+
+
+# Check MSVC Architechture
+string(COMPARE EQUAL "@HUNTER_MSVC_ARCH@" "x86"   _is_arch_x86)
+string(COMPARE EQUAL "@HUNTER_MSVC_ARCH@" "amd64" _is_arch_x64)
+
+if(_is_arch_x86)
+    set(msvc_arch "Win32")
+elseif(_is_arch_x64)
+    set(msvc_arch "x64")
+else()
+    hunter_user_error("Architechture supplied is not supported.")
+endif()
+
+
+#Check Configuration Types
+string(TOUPPER @HUNTER_PACKAGE_CONFIGURATION_TYPES@ configuration_type_upper)
+string(COMPARE EQUAL ${configuration_type_upper} "RELEASE" _is_release)
+string(COMPARE EQUAL ${configuration_type_upper} "DEBUG"   _is_debug)
+if(_is_release)
+    set(configuration_type "Release")
+elseif(_is_debug)
+    set(configuration_type "Debug")
+else()
+    hunter_user_error("Invalid build configuration type provided (HUNTER_PACKAGE_CONFIGURATION_TYPES). Valid configuration types are: Release or Debug")
+endif()
+
+
+# Check Shared Lib
+if(BUILD_SHARED_LIBS)
+    string(CONCAT ${configuration_type} ${configuration_type} "DLL")
+endif()
+
+
+# Print MSVC Values
+if(HUNTER_STATUS_DEBUG)
+    hunter_status_debug("------- libsodium MSBuild values -------")
+    hunter_status_debug("Platform Toolset:            ${msvc_platform_toolset}")
+    hunter_status_debug("Architechture:               ${msvc_arch}")
+    hunter_status_debug("Build Configuration:         @HUNTER_PACKAGE_CONFIGURATION_TYPES@")
+    hunter_status_debug("MSBuild Build Configuration: ${configuration_type}")
+endif()
+
+file(TO_NATIVE_PATH "@HUNTER_PACKAGE_SOURCE_DIR@\\libsodium.vcxproj" proj_file_path)
+
+# ExternalProject
+ExternalProject_Add("@HUNTER_EP_NAME@"
+    URL
+        @HUNTER_PACKAGE_URL@
+    URL_HASH
+        SHA1=@HUNTER_PACKAGE_SHA1@
+    DOWNLOAD_DIR
+        "@HUNTER_PACKAGE_DOWNLOAD_DIR@"
+    SOURCE_DIR
+        "@HUNTER_PACKAGE_SOURCE_DIR@"
+    INSTALL_DIR
+        "@HUNTER_PACKAGE_INSTALL_PREFIX@"
+        # not used, just avoid creating Install/<name> empty directory
+    CONFIGURE_COMMAND
+        ""
+    BUILD_COMMAND
+        msbuild ${proj_file_path} /p:PlatformToolset=${msvc_platform_toolset} /p:Configuration=${configuration_type} /p:PostBuildEventUseInBuild=false
```

Comment:
There are different versions (version numbers) but they wont break anything, as long as the msbuild from VC is used/installed it will work